charts/gha-runner-scale-set/templates/autoscalingrunnerset.yaml

```diff
@@ -11,6 +11,11 @@ metadata:
   namespace: {{ .Release.Namespace }}
   labels:
     {{- include "gha-runner-scale-set.labels" . | nindent 4 }}
+  annotations:
```

Comment:
should we just put all resource names with the cleanup finalizer as annotations, so you don't need to re-construct the name in the resourceBuilder?